controllers/btpoperator_controller.go

```diff
@@ -219,34 +224,15 @@ func (r *BtpOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	if err := r.Get(ctx, req.NamespacedName, reconcileCr); err != nil {
 		if k8serrors.IsNotFound(err) {
 			logger.Info(fmt.Sprintf("%s BtpOperator CR not found. Ignoring it since object has been deleted.", req.Name))
-			existingBtpOperators := &v1alpha1.BtpOperatorList{}
-			if err := r.List(ctx, existingBtpOperators); err != nil {
-				logger.Error(err, "unable to get existing BtpOperator CRs")
-				return ctrl.Result{}, nil
-			}
-			if len(existingBtpOperators.Items) > 0 {
-				return r.setNewLeader(ctx, existingBtpOperators)
-			}
-
 			return ctrl.Result{}, nil
 		}
 		logger.Error(err, "unable to get BtpOperator CR")
 		return ctrl.Result{}, err
 	}
 
-	existingBtpOperators := &v1alpha1.BtpOperatorList{}
-	if err := r.List(ctx, existingBtpOperators); err != nil {
-		logger.Error(err, "unable to get existing BtpOperator CRs")
-		return ctrl.Result{}, err
-	}
-
-	if len(existingBtpOperators.Items) > 1 {
-		oldestCr := r.getOldestCR(existingBtpOperators)
-		if reconcileCr.GetUID() == oldestCr.GetUID() {
-			reconcileCr.Status.Conditions = nil
-		} else {
-			return ctrl.Result{}, r.HandleRedundantCR(ctx, oldestCr, reconcileCr)
-		}
+	if req.Name != btpoperatorCRName || req.Namespace != kymaSystemNamespaceName {
+		logger.Info(fmt.Sprintf("BtpOperator CR %s/%s is not the one we are looking for. Ignoring it.", req.Namespace, req.Name))
+		return ctrl.Result{}, r.HandleWrongNamespaceOrName(ctx, reconcileCr)
```

Comment:
`HandleWrongNamespaceOrName` still reconciles BtpOperator CRs with a different name or in a different namespace. I think this is a good reason to keep the tests for reconciliation of CRs with wrong name or in a wrong namespace.

Reply:
added unit tests: https://github.com/kyma-project/btp-manager/pull/1027/files#diff-0cbf4582156f82c4d2da3f09f2e7dd36e465da0b9ecd294709d412dc84bafa6dR78

modified e2e tests: https://github.com/kyma-project/btp-manager/pull/1027/files#diff-1a49fddd7356092fd1664e0521996abc8618cf484c7193a872662c4b9083daa2R81